src/Features/LanguageServer/Protocol/Features/Options/SolutionCrawlerOptionsStorage.cs

```diff
@@ -48,4 +56,36 @@ public static BackgroundAnalysisScope GetBackgroundAnalysisScope(this IGlobalOpt
         return globalOptions.GetOption(SolutionBackgroundAnalysisScopeOption) ??
                globalOptions.GetOption(BackgroundAnalysisScopeOption, language);
     }
+
+    public static bool IsFullSolutionAnalysisEnabled(this DiagnosticAnalyzer analyzer, IGlobalOptionService globalOptions, string language)
+    {
+        if (analyzer.IsCompilerAnalyzer())
+        {
+            return globalOptions.GetOption(CompilerDiagnosticsScopeOption, language) == CompilerDiagnosticsScope.FullSolution;
+        }
+
+        return GetBackgroundAnalysisScope(globalOptions, language) == BackgroundAnalysisScope.FullSolution;
+    }
+
+    public static bool IsFullSolutionAnalysisEnabled(
+        this IGlobalOptionService globalOptions,
+        string language,
+        out bool compilerFullSolutionAnalysisEnabled,
+        out bool analyzersFullSolutionAnalysisEnabled)
+    {
+        compilerFullSolutionAnalysisEnabled = globalOptions.GetOption(CompilerDiagnosticsScopeOption, language) == CompilerDiagnosticsScope.FullSolution;
+        analyzersFullSolutionAnalysisEnabled = GetBackgroundAnalysisScope(globalOptions, language) == BackgroundAnalysisScope.FullSolution;
+        return compilerFullSolutionAnalysisEnabled || analyzersFullSolutionAnalysisEnabled;
+    }
+
+    public static bool IsAnalysisDisabled(
+        this IGlobalOptionService globalOptions,
+        string language,
+        out bool compilerDiagnosticsDisabled,
+        out bool analyzersDisabled)
+    {
+        compilerDiagnosticsDisabled = globalOptions.GetOption(CompilerDiagnosticsScopeOption, language) == CompilerDiagnosticsScope.None;
+        analyzersDisabled = GetBackgroundAnalysisScope(globalOptions, language) == BackgroundAnalysisScope.None;
+        return compilerDiagnosticsDisabled && analyzersDisabled;
```

Comment:
not sure i understand this.  say teh analysis scope is something like "open files", dont' we need to check for openness here?

Reply:
So this API is meant to check if background analysis is completely disabled for both compiler diagnostics and analyzer diagnostics (seems highly unlikely someone who do that, but those who like notepad kind editing experience, they can). If either one of them is enabled for any scope, then some background analysis will happen for diagnostics and this API will return false. Let me add a doc comment here.